java/ql/test/library-tests/frameworks/JaxWs/JakartaRs4.java

```diff
@@ -0,0 +1,106 @@
+import java.io.InputStream;
+import java.io.IOException;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.DELETE;
+import jakarta.ws.rs.PUT;
+import jakarta.ws.rs.OPTIONS;
+import jakarta.ws.rs.HEAD;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.BeanParam;
+import jakarta.ws.rs.CookieParam;
+import jakarta.ws.rs.FormParam;
+import jakarta.ws.rs.HeaderParam;
+import jakarta.ws.rs.MatrixParam;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.QueryParam;
+import jakarta.ws.rs.client.Client;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.MultivaluedMap;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.ext.MessageBodyReader;
+
+// This is not a resource class because it doesn't have a @Path annotation.
+// Note that inheritance of class or interface annotations is not supported in
+// JAX-RS.
+public class JakartaRs4 implements JakartaRsInterface { // $ RootResourceClass
```

Comment:
It seems a bit confusing to have a comment saying that this is not a resource class along with a `// $ RootResourceClass` tag. Would it make sense to reword the comment to something like the below?

```suggestion
// By the JAX-RS spec, this is not a resource class because it doesn't 
// have a @Path annotation. Inheritance of class or interface annotations 
// is not supported in JAX-RS. However, this is a resource class for some 
// implementations, like Apache CXF, that allow inheritance of JAX-RS 
// annotations.
public class JakartaRs4 implements JakartaRsInterface { // $ RootResourceClass
```